tests/php/core-api/wpcom-fields/test-post-fields-publicize-connections.php

```diff
@@ -38,6 +38,29 @@ public static function wpSetUpBeforeClass( $factory ) {
 
 		self::$user_id = $factory->user->create( array( 'role' => 'administrator' ) );
 
+		// Hacky :(
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			self::setup_connections_wpcom();
+		} else {
+			self::setup_connections_jetpack();
+		}
+	}
+
+	static function setup_connections_wpcom() {
+		global $wpdb;
+
+		$wpdb->insert( 'external_access_tokens', array( 'user_id' => self::$user_id, 'external_display' => 'test-display-name456', 'token' => 'fb-token', 'provider' => 'facebook' ) );
+		$keyring_token_id = (string) $wpdb->insert_id;
+		self::$connection_ids[] = $keyring_token_id;
+		$wpdb->insert( 'publicize_connections', array( 'user_id' => self::$user_id, 'token_id' => $keyring_token_id, 'blog_id' => get_current_blog_id() ) );
+
+		$wpdb->insert( 'external_access_tokens', array( 'user_id' => self::$user_id, 'external_display' => 'test-display-name123', 'token' => 't-token', 'provider' => 'tumblr' ) );
+		$keyring_token_id = (string) $wpdb->insert_id;
+		self::$connection_ids[] = $keyring_token_id;
+		$wpdb->insert( 'publicize_connections', array( 'user_id' => 0 /* global connection */, 'token_id' => $keyring_token_id, 'blog_id' => get_current_blog_id() ) );
+	}
+
+	static function setup_connections_jetpack() {
```

Comment:
Definitely not a requirement for this PR, but it would be lovely if we share these common helpers in this tests class and the inactive one.

Reply:
That's a good call! Although I'd rather not do this. This file will be fusion synced with WPCOM (and `-inactive` will not), with your suggestion we would need to add one more file to the list, which may or may not complicate things and I not sure will it simplify things either. I will be happy to DRY this function once we see increased usage of it.